### PR TITLE
Update classnames in Pagination component

### LINKED ISSUE #4257: `p-pagination` class name should be on nav element
#### [Pagination component](https://vanillaframework.io/docs/patterns/pagination)
After the recent addition of the `<nav>` element to the pagination component, there should be a re-shuffle of the classnames. Similar to the side navigation, the `p-navigation` class should be moved to the `nav` element, and there should be an additional class added to the `<ol>` element, maybe `p-pagination__items`?

### PR DESCRIPTION
**Pull request by bethcollins92**
## Done

- Move `p-pagination` classname to the `<nav>` element
- Add new `p-pagination__items` class name to the `<ol>` element. 

It seems the styles can be applied to both class names and nothing breaks on old and new structure... 

Fixes #4257 

## QA

- Open [demo](https://vanilla-framework-4451.demos.haus)
- Check all examples of Pagination component. Update the HTML in the browser to the older structure to check it's backwards compatible
- Review updated documentation:
  - What's new table
  - Side nav in docs should be updated

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical-web-and-design/vanilla-framework/blob/main/templates/docs/whats-new.md).
- [x] Documentation side navigation should be updated with the relevant labels.

### CODE REVIEW

**Comment by bartaz** [comment]
We should mention in the docs, what's new page and in code comments that using `.p-pagination` on ul element is deprecated, and `p-pagination` should be used on nav element with `p-pagination__items` in it.

File: scss/_patterns_pagination.scss [hunk 1]
```diff
@@ -13,7 +13,8 @@
     }
   }
 
-  .p-pagination {
+  .p-pagination,
```
Comment (bartaz) [inline scss/_patterns_pagination.scss:16]: I think we should by fine by doing `ul.p-pagination`, `ol.p-pagination` here to make sure it's only applied when needed.

But a comment in code should be added to mention that this is deprecated.

File: templates/docs/whats-new.md [hunk 2]
```diff
@@ -20,7 +20,31 @@ When we add, make significant updates, or deprecate a component we update their
     </tr>
   </thead>
   <tbody>
-  <!-- 3.3.0 -->
+ <!-- 3.4.0 -->
```
Comment (bartaz) [inline templates/docs/whats-new.md:23]: The deprecated use of p-pagination class name should be listed here as well for future reference. Thanks!

**Comment by bethcollins92** [comment]
@bartaz Made the suggested additions, _should_ be all good now

**Comment by bethcollins92** [comment]
@bartaz The `ol.p-pagination` is breaking the github check

**Comment by bartaz** [comment]
@bethcollins92 the error is in linting, not build, you can prevent this by using `// stylelint-disable selector-max-type` - you can find some examples in other places already

**Comment by anthonydillon** [comment]
You could be more explicit and use `.p_navigation:not(nav)` for the depreciation selector.

**Comment by bartaz** [comment]
> You could be more explicit and use `.p_navigation:not(nav)` for the depreciation selector.

That's a good call, thanks!

**Comment by bethcollins92** [comment]
Thanks for that @anthonydillon and @bartaz - should be ready for review

**Review by bartaz — APPROVED** [review]
LGTM

**Pull request merged**